src/common/include/gudhi/random_graph_generators.h

```diff
@@ -0,0 +1,98 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Vincent Rouvreau
+ *
+ *    Copyright (C) 2024 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#ifndef RANDOM_GRAPH_GENERATORS_H_
+#define RANDOM_GRAPH_GENERATORS_H_
+
+#include <gudhi/Random.h>
+
+
+#include <algorithm>  // for std::prev_permutation
+#include <vector>
+#include <array>
+#include <random>
+#include <numeric>  // for std::iota
+#include <cstddef>  // for std::size_t
+#include <type_traits>  // for std::is_floating_point_v
+#ifdef DEBUG_TRACES
+#include <iostream>
+#endif  // DEBUG_TRACES
+
+std::random_device rd;
+
+namespace Gudhi {
+
+template <typename Vertex_handle>
+std::vector<std::array<Vertex_handle, 2>> random_edges(Vertex_handle nb_vertices, double density = 0.15) {
+  std::vector<std::array<Vertex_handle, 2>> permutations;
+  if (nb_vertices < 2)
+    return permutations;
+
+  std::vector<bool> to_permute(nb_vertices);
+  std::fill(to_permute.begin(), to_permute.begin() + 2, true);
+  
+  std::size_t nb_permutations = (nb_vertices * (nb_vertices - 1)) / 2;
+  auto random_values = Gudhi::Random().get_range<double>(nb_permutations);
```

Comment:
If you are going to create Random objects left and right, I don't see how this will help determinism. There is no way for the user to specify a seed...